### DISCUSSION TITLE
LAYOUT.VERSION.MINOR error

### DISCUSSION THREAD

**Discussion by mthsdiniz-usp** [pyproj4/pyproj#1465] [Q&A] [answered]
Hello all, i've been trying to run an Insar analysis using a package called dolphin (https://github.com/isce-framework/dolphin) but i've been having a few issues when trying to setup the environment.

I'm currently using AWS Sagemaker Classic Studio and setting it up using the following simple docker file:

```
FROM public.ecr.aws/ubuntu/ubuntu:20.04

RUN \
  apt-get update && \
  apt-get install -y wget nano

# installing miniconda
RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
RUN bash Miniconda3-latest-Linux-x86_64.sh -b -p /miniconda
ENV PATH=$PATH:/miniconda/condabin:/miniconda/bin

# install libraries for the base environment
RUN conda install numpy scikit-learn ipykernel gdal pyproj 
```

At first, pyproj was not recognizing the proj.db, so i had to setup an environment variable in my jupyter notebook:

```
import pyproj
os.environ['PROJ_LIB'] = pyproj.datadir.get_data_dir()
```

Right now when executing the package i'm getting an error related to the LAYOUYT.VERSION.MINOR of pyproj, which is issuing the following message:

![image](https://github.com/user-attachments/assets/91819d85-a286-4e7a-878d-2cc1ee8e6a10)

Does anyone experienced it before or knows how to deal with this type of problems?

ps: i'm installing dolphin using pip and it installs rasterio as one of the requirements... Can it be also installing pyproj in the back and creating multiple version of it?

Thanks!

**Comment by snowman2** [comment, marked as answer]
It is recommended to avoid using pip install with a conda environment as it often causes issues and unexpected behavior: https://pyproj4.github.io/pyproj/stable/installation.html
